### PR TITLE
K-allagbe/issue18-update-env-var-template

### LINKED ISSUE #18: Update Environment Variables Template Mismatch and Other Issues in finesse-backend
We need to address a few lingering issues in `finesse-backend` that weren't handled in previous PRs.

1. **Mismatched Environment Variable in Template**: The environment variable in the code was renamed to `FINESSE_BACKEND_STATIC_FILE_URL`, but the `env.template` still shows the old name, `FINESSE_BACKEND_GITHUB_STATIC_FILE_URL`.

2. **Missing Global Error Handling in Flask App**: Our Flask app currently lacks a global error handler. We need to implement this to manage route errors better. Also, when in debug mode, Flask should provide detailed error information, which it isn't doing at the moment.

3. **Out-of-Date Documentation**: The project documentation hasn't been updated to reflect the latest changes and needs a refresh.

4. **Incorrect Naming of Devcontainer Folder**: The devcontainer configuration folder is named `devcontainer` instead of `.devcontainer`, which is the standard.

### Tasks:
- [ ] Update `env.template` to reflect the correct environment variable name.
- [ ] Implement a global error handler in the Flask application.
- [ ] Adjust Flask settings to provide detailed error information in responses when debug mode is on, while properly logging the errors.
- [ ] Update the project documentation to align with recent changes.
- [ ] Rename the devcontainer configuration folder to `.devcontainer`.
- [ ] Write tests to ensure the new error handling works as expected.

### Acceptance Criteria:
- The `env.template` should use the updated environment variable name.
- The Flask app should have a working global error handler.
- Detailed error messages are returned in responses when Flask is in debug mode.
- The errors are properly logged
- The documentation accurately represents the current state of the project.
- The devcontainer configuration folder is correctly named `.devcontainer`.
- All tests pass successfully.

### PR DESCRIPTION
**Pull request by k-allagbe**
Closes #18
- [x] update `.env.template`
- [x] global error handling
- [x] unit tests
- [x] update doc

### CODE REVIEW

File: app/blueprints/common.py
Comment (k-allagbe) [reply app/blueprints/common.py:14]: This is conditional to the debug mode. We want to return the trace in debug mode.

**Comment by k-allagbe** [comment]
@redat97 There seems to be no lint-and-test workflow for this repo?

Also here:
https://github.com/ai-cfia/finesse-backend/blob/8e8a8b5be0467e8c7b925dd7647ba7d885fd0bdb/.github/workflows/azure-deployment.yml#L10
It seems we are not using the main branch?

**Comment by redat97** [comment]
> @redat97 There seems to be no lint-and-test workflow for this repo?
> 
> Also here:
> 
> https://github.com/ai-cfia/finesse-backend/blob/8e8a8b5be0467e8c7b925dd7647ba7d885fd0bdb/.github/workflows/azure-deployment.yml#L10
> 
> 
> It seems we are not using the main branch?

it's not using the main branch because some changes were needed that's are not approved on the main yet, it will work fine either way since I don't delete the branch after I merge.

As for lint-test, I think it may be better to add it to a separate issue unless you know they will pass because if you have to add a bunch of commits refering to another issue, it might be best to have it in a separate PR, let me know and I will add them to this one if needed

**Comment by k-allagbe** [comment]
> > @redat97 There seems to be no lint-and-test workflow for this repo?
> > Also here:
> > https://github.com/ai-cfia/finesse-backend/blob/8e8a8b5be0467e8c7b925dd7647ba7d885fd0bdb/.github/workflows/azure-deployment.yml#L10
> > 
> > It seems we are not using the main branch?
> 
> it's not using the main branch because some changes were needed that's are not approved on the main yet, it will work fine either way since I don't delete the branch after I merge.
> 
> As for lint-test, I think it may be better to add it to a separate issue unless you know they will pass because if you have to add a bunch of commits refering to another issue, it might be best to have it in a separate PR, let me know and I will add them to this one if needed

Got you. Your call, really. I'm just drawing your attention to this.

**Comment by k-allagbe** [comment]
@rngadam we already had a conversation about https://github.com/ai-cfia/finesse-backend/security/code-scanning/5 here: https://github.com/ai-cfia/finesse-backend/pull/7#discussion_r1404661956 where we decided to sanitize the user inputs as close as possible to the entry point. The security bot doesn't seem to recognize this for some reason. Should we dismiss it as not valid or take a closer look?

**Comment by rngadam** [comment]
> @rngadam we already had a conversation about [Log Injection](/ai-cfia/finesse-backend/security/code-scanning/5) here: [#7 (comment)](https://github.com/ai-cfia/finesse-backend/pull/7#discussion_r1404661956) where we decided to sanitize the user inputs as close as possible to the entry point. The security bot doesn't seem to recognize this for some reason. Should we dismiss it as not valid or take a closer look?

dismissed with explanation

**Review by rngadam — APPROVED** [review]

**Comment by rngadam** [comment]
@k-allagbe need to resolve conflicts

**Review by rngadam — APPROVED** [review]

**Pull request merged**
